### PR TITLE
Fixed deletion of wrong session duration item

### LINKED ISSUE #63: Wrong session duration item deleted
### Problem
In the Nedit Download page, in the Session Lengths tab, the trash icon for the list of session duration/price combos does not delete the session duration being clicked on. Rather, it deletes the first session duration in the list, no matter which trash icon is clicked.

### Steps to reproduce
1. Open some download with the list of session durations.
2. Click on trash icon near last one.
3. First item will be removed.

### Reason
Session durations come without id from backend. We rely on some ID during removing.

### Solution
Generate ID for items from backend and remove that ID before sending to backend.

### PR DESCRIPTION
**Pull request by romalytvynenko**
fixes #63

### CODE REVIEW

File: src/components/SessionLength.js [hunk 1]
```diff
@@ -151,7 +151,7 @@ export default function CfSessionLength (Vue, Vuex, FunctionalArrayCollection) {
         const sessionLength = this.sessionTimeUnit * this.sessionDefault.sessionLength
 
         this.sessions.addItem({
-          id: this.sessions.getItems().length,
+          id: Math.random().toString(36).substring(7),
```
Comment (XedinUnknown) [inline src/components/SessionLength.js]: Why generate a random ID? The session already has unique data that can be used. Or can the same session length appear more than once?

Comment (romalytvynenko) [reply src/components/SessionLength.js]: Hmm... You are right! We have ability to use session length as unique ID. For now we can duplicate session lengths, but this has no sense, so I will disallow it and show message about it.

File: src/components/SessionLength.js [hunk 2]
```diff
@@ -146,34 +146,56 @@ export default function CfSessionLength (Vue, Vuex, FunctionalArrayCollection) {
        * can keep adding new sessions.
        */
       addNewSession () {
-        if (!this.validate()) return;
+        const normalizedSession = this.normalizeSessionLength(this.sessionDefault)
 
-        const sessionLength = this.sessionTimeUnit * this.sessionDefault.sessionLength
+        if (!this.validate(normalizedSession)) return
```
Comment (XedinUnknown) [inline src/components/SessionLength.js]: How will we know if there's a problem with the session? For example, in the future, when we have notifications implemented, how can we modify this to add an `error` type notification to the stack, with the text of the validation error.

Comment (romalytvynenko) [reply src/components/SessionLength.js]: For now this method will validate session length and display error in UI, if there was something wrong. In the future we can easily throw `Errors` and handle it somehow.

Comment (XedinUnknown) [reply src/components/SessionLength.js]: Ah, so `validate()` is something that is responsible for the UI part as well?

Comment (romalytvynenko) [reply src/components/SessionLength.js]: Yes 👌

File: src/components/SessionLength.js [hunk 3]
```diff
@@ -146,34 +146,56 @@ export default function CfSessionLength (Vue, Vuex, FunctionalArrayCollection) {
...
+       * @param {SessionLength} sessionLength Session length to validate.
+       *
        * @return {boolean}
        */
-      validate () {
-        if (Number(this.sessionDefault.sessionLength) <= 0) {
+      validate (sessionLength) {
+        const hasSameSession = this.sessions.hasItem(sessionLength)
+        if (hasSameSession) {
+          this.validationError = this.$refs.sessionLength.dataset.uniqnessError
```
Comment (XedinUnknown) [inline src/components/SessionLength.js]: Couldn't really understand where the errors are coming from. But we can sort this out later.

File: src/transformers/SessionLengthReadTransformer.js [hunk 4]
```diff
@@ -13,6 +13,9 @@ export default class SessionLengthReadTransformer extends Transformer {
    */
   rules = {
     price: (model) => {
+      if (!model['id']) {
+        model['id'] = '_' + Math.random().toString(36).substring(7)
```
Comment (XedinUnknown) [inline src/transformers/SessionLengthReadTransformer.js]: Looks like session length IDs are still random here.

File: src/components/SessionLength.js (hunk 3 above)
Comment (romalytvynenko) [reply src/components/SessionLength.js]: This is nice part. Errors are coming from backend's templates.

File: src/transformers/SessionLengthReadTransformer.js (hunk 4 above)
Comment (romalytvynenko) [reply src/transformers/SessionLengthReadTransformer.js]: Thanks, you are right. Fixing.

File: src/components/SessionLength.js (hunk 3 above)
Comment (XedinUnknown) [reply src/components/SessionLength.js]: Isn't i18n already effectively the same thing, if the backend provides translations?

Comment (romalytvynenko) [reply src/components/SessionLength.js]: 👍

**Pull request merged**